src/include/sof/schedule/ll_schedule_source.h

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Tomasz Lauda <tomasz.lauda@linux.intel.com>
+ */
+
+#ifndef __SOF_SCHEDULE_LL_SCHEDULE_SOURCE_H__
+#define __SOF_SCHEDULE_LL_SCHEDULE_SOURCE_H__
+
+#include <sof/atomic.h>
+#include <sof/debug/panic.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/clk.h>
+#include <sof/spinlock.h>
+#include <sof/trace/trace.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+struct ll_schedule_source;
+struct task;
+
+/* tracing */
+#define trace_sch_src(format, ...) \
+	trace_event(TRACE_CLASS_LL_SCHED_SRC, format, ##__VA_ARGS__)
+
+#define trace_sch_src_error(format, ...) \
+	trace_error(TRACE_CLASS_LL_SCHED_SRC, format, ##__VA_ARGS__)
+
+#define tracev_sch_src(format, ...) \
+	tracev_event(TRACE_CLASS_LL_SCHED_SRC, format, ##__VA_ARGS__)
+
+struct ll_schedule_source_ops {
+	int (*source_register)(struct ll_schedule_source *src, uint64_t period,
```

Comment:
So "source" is the scheduling time domain ? I would rename to "domain" as source and SRC overlap with buffers and SRCs.

Reply:
Good suggestion.
Done.